ocs_ci/deployment/baremetal.py

```diff
@@ -11,3 +21,90 @@ class BAREMETALUPI:
     def __init__(self):
         logger.info("BAREMETAL UPI")
         # TODO: OCP,OCS deployment
+
+
+class BAREMETALPSIUPI(Deployment):
+    """
+    All the functionalities related to BaremetalPSI- UPI deployment
+    lives here
+    """
+    def __init__(self):
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        super().__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super().__init__()
+            self.flexy_instance = FlexyBaremetalPSI()
+            self.build_psi_conf()
+            self.utils = psiutils.PSIUtils(self.psi_conf)
+
+        def deploy_prereq(self):
+            """
+            Instantiate proper flexy class here
+
+            """
+            super().deploy_prereq()
+            self.flexy_instance.deploy_prereq()
+
+        def deploy(self):
+            self.flexy_instance.deploy()
+            # Point cluster_path to cluster-dir created by flexy
+            abs_cluster_path = os.path.abspath(self.cluster_path)
+            flexy_cluster_path = os.path.join(
+                self.flexy_instance.flexy_mnt_host_dir,
```

Comment:
This was fun!!
Actually  having flexy directory on ~/current-cluster-dir is a problem, container will not be able to start because it tries to do some modifications for flexy dir on nfs share and the op fails on nfs.

I tried having flexy dir on Jenkins slave's local dir and making ~/current-cluster-dir/openshift-cluster-dir as a softlink to this dir and container works fine and it installed.

Only think I need to check is ,this soft link doesn't cause any problem for OCS deployment.

@petr-balogh , Are you ok with this approach ?

Reply:
What about cloning the private repo to `ocs-ci/data` directory? (There is already `rook`...)

Having `~/current-cluster-dir/openshift-cluster-dir` softlink  pointing to the clusterdir generated by flexy is not ideal, because it will not be backed up on the NFS. From my POV, it will be better to have the `flexy` directory (somewhere) in `data` directory of the `ocs-ci` repo directory and prospectively copy all the content somewhere to the defined cluster dir.

Reply:
Making ```ocs-ci/data``` as the initial flexy working dir and then copying to nfs mount works. 
@petr-balogh I remember from our discussion that copying to NFS mount should be done from Jenkins code rather than OCS-CI , shall we stick to same plan ? I haven't added any copying code in updated PR .

Reply:
@shylesh what I remember that we discussed you will cover copy here from ocs-ci .  In the jenkins we can do some second check if the execution was aborted, then we can do the copy from jenkins as well if it was not completed by ocs-ci.    So please go with copy from ocs-ci and from the jenkins job we can do later on to be sure we did not loose any data.

Reply:
@petr-balogh sorry I might have missed that part, ack, I will do it in ocs-ci as well.